x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlCompatIT.java

```diff
@@ -174,11 +179,10 @@ private void assertCursorNotCompatibleAcrossVersions(Version version1, RestClien
         indexDocs();
 
         Request req = new Request("POST", "_sql");
-        // GROUP BY queries always return a cursor
-        req.setJsonEntity(sqlQueryEntityWithOptionalMode("SELECT int FROM test GROUP BY 1", bwcVersion));
+        req.setJsonEntity(sqlQueryEntityWithOptionalMode(Map.of("query", "SELECT int FROM test", "fetch_size", 1), bwcVersion));
         Map<String, Object> json = performRequestAndReadBodyAsJson(client1, req);
         String cursor = (String) json.get("cursor");
-        assertThat(cursor, Matchers.not(Matchers.emptyString()));
+        assertThat(cursor, Matchers.not(Matchers.emptyOrNullString()));
```

Comment:
the missing null check caused the `null` cursor to be propagated to the next request resulting in a `"null"` cursor.